RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -5792,3 +5797,140 @@ int auto_reserveCraftAmount(item orig_it)
 	}
 	return inner(orig_it);
 }
+
+
+// This just does the math for comparing vs. the Cap. If no cap is set then ML is virtually unlimited.
+int remainingMLToCap()
+{
+	int MLToCap = 0;
+
+	if(get_property("auto_MLSafetyLimit") == "")
+	{
+		MLToCap = 999999;
+	}
+	else if(monster_level_adjustment() < get_property("auto_MLSafetyLimit"))
+	{
+		MLToCap = get_property("auto_MLSafetyLimit") - monster_level_adjustment();
+	}
+
+	return MLToCap;
+}
+
+// Gives us the number we need when comparing to a desired ML or entering a value into a maximizer string.
+int auto_convertDesiredML(int DML)
+{
+
+	int DesiredML = get_property("auto_MLSafetyLimit");
+
+	if((get_property("auto_MLSafetyLimit") == "") || (DML < get_property("auto_MLSafetyLimit")))
+	{
+		DesiredML = DML;
+	}
+		
+	return DesiredML;
+}
+
+// Uses MCD in the constraints of a Cap
+boolean auto_setMCDToCap()
+{
+	if(($strings[Marmot, Opossum, Platypus] contains my_sign()) && (11 <= remainingToMLCap()))
+	{
+		auto_change_mcd(11);
+	}
+	else if(10 <= remainingToMLCap())
+	{
+		auto_change_mcd(10);
+	}
+	else if(10 > remainingToMLCap())
+	{
+		auto_change_mcd(remainingToMLCap());
+	}
+
+	return true;
+}
+
+// We use this function to determine the suitability of using Ur-Kel's
+boolean UrKelCheck(int UToML, int UUL, int ULL)
```

Comment:
Nit pick (verging on asking for a change): its generally good coding practice to spell out variable names explicitly. I have no idea what these parameters mean without guessing and reading the code to understand how they are used. You might not have even written this, acronym and abbreviated variable names all over the place. And for what? copy+paste works equally fast for a 4 letter acronym as it does for a sentence length variable name that clearly spells out what its for. Readability >>>>>> most things. Sorry, getting off my soap box now.

Reply:
Done.

You are right it is a better practice.